x-pack/heartbeat/monitors/browser/synthexec/enrich.go

```diff
@@ -112,6 +114,9 @@ func (je *journeyEnricher) enrichSynthEvent(event *beat.Event, se *SynthEvent) e
 	}
 
 	switch se.Type {
+	case "cmd/status":
```

Comment:
We should guard this summary creation event. If the user has afterAll hook that throws error, we would have both `journey/end` followed by `cmd/status`.

Reply:
This is a sort of 'task failed successfully' situation. I guess there's a question here, does a failing after hook invalidate the test result at all? It feels weird to, say, trigger an alert based on an `afterAll` hook. 

I'm wondering if we really would need to introduce a new concept of warnings into the schema / UI. Thoughts?

Reply:
I don't think so, Lots of test runners don't consider failures in the afterAll hooks to be failures. They mainly exist to keep some cleanup mechanism. But having warnings in the UI sounds like a cool idea.

Reply:
The `afterAll` hooks question is a tricky one. I previously did some work on Jest's hooks here (https://github.com/facebook/jest/pull/8654) where this same discussion has come up.

Given how complex the discussion on hooks semantics is, we didn't get to a conclusion in that PR (which is why it's still open and not much has happened since then).

Anyway, I do feel like failures in `afterHooks` should be test failures. If you think about how hooks could be implemented using a procedural API under the hood, they end up being part of the upper test scope, and thus would trigger failures. For inspiration, see Deno's RFC on their testing API: https://github.com/denoland/deno/discussions/10771.

(That's just for further discussion as I'd consider that to be out of the scope of that PR if you agree)

Reply:
> I'd consider that failing a journey would be out of the scope of this PR if you agree.

++, Lets do this separately and open a issue in Synthetics runner. 

> Anyway, I do feel like failures in afterHooks should be test failures.

Mainly, in the runner i didn't go with this idea as we didn't have a good way of reporting the error. `Warnings` would make a lot of sense here.